src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -416,11 +416,11 @@ public IEnumerable<Document> GetDocumentsStartingWith(DocumentsOperationContext
                     }
                     docCount++;
                     var document = TableValueToDocument(context, ref result.Reader);
-                    string documentKey = document.Key;
-                    if (documentKey.StartsWith(prefix, StringComparison.CurrentCultureIgnoreCase) == false)
+                    
+                    if (document.Key.StartsWith(prefix) == false)
```

Comment:
so we need to materialize `string` anyway. closing PR.  BTW I don't think it should be culture case sensitive, should it?